sky/engine/bindings/mojo_services.cc

```diff
@@ -20,87 +20,97 @@ MojoServices* GetMojoServices() {
+  Dart_SetIntegerReturnValue(args, GetMojoServices()->TakeShell());
 }
 
-void DartTakeViewHandle(Dart_NativeArguments args) {
-  Dart_SetIntegerReturnValue(
-      args, GetMojoServices()->TakeViewHandle().value());
+void DartTakeView(Dart_NativeArguments args) {
+  Dart_SetIntegerReturnValue(args, GetMojoServices()->TakeView());
+}
+
+void DartTakeViewServices(Dart_NativeArguments args) {
+  Dart_SetIntegerReturnValue(args, GetMojoServices()->TakeViewServices());
 }
 
 }  // namespace
 
 void MojoServices::RegisterNatives(DartLibraryNatives* natives) {
   natives->Register({
-    {"takeRootBundleHandle", DartTakeRootBundleHandle, 0, true},
-    {"takeServicesProvidedByEmbedder", DartTakeServicesProvidedByEmbedder, 0, true},
-    {"takeServicesProvidedToEmbedder", DartTakeServicesProvidedToEmbedder, 0, true},
-    {"takeShellProxyHandle", DartTakeShellProxyHandle, 0, true},
-    {"takeViewHandle", DartTakeViewHandle, 0, true},
+    {"MojoServices_takeRootBundle", DartTakeRootBundle, 0, true},
+    {"MojoServices_takeIncomingServices", DartTakeIncomingServices, 0, true},
+    {"MojoServices_takeOutgoingServices", DartTakeOutgoingServices, 0, true},
+    {"MojoServices_takeShell", DartTakeShell, 0, true},
+    {"MojoServices_takeView", DartTakeView, 0, true},
+    {"MojoServices_takeViewServices", DartTakeViewServices, 0, true},
   });
 }
 
 void MojoServices::Create(Dart_Isolate isolate,
                           sky::ServicesDataPtr services,
-                          mojo::ServiceProviderPtr services_from_embedder,
+                          mojo::ServiceProviderPtr incoming_services,
                           mojo::asset_bundle::AssetBundlePtr root_bundle) {
   FlutterDartState* state = static_cast<FlutterDartState*>(
       DartState::From(isolate));
   state->set_mojo_services(std::unique_ptr<MojoServices>(new MojoServices(
-      services.Pass(), services_from_embedder.Pass(), root_bundle.Pass())));
+      services.Pass(), incoming_services.Pass(), root_bundle.Pass())));
 }
 
 MojoServices::MojoServices(sky::ServicesDataPtr services,
-                           mojo::ServiceProviderPtr services_from_embedder,
+                           mojo::ServiceProviderPtr incoming_services,
                            mojo::asset_bundle::AssetBundlePtr root_bundle)
   : services_(services.Pass()),
-    services_from_embedder_(services_from_embedder.Pass()),
-    root_bundle_(root_bundle.Pass()) {
-  if (services_ && services_->services_provided_to_embedder.is_pending()) {
-    services_provided_to_embedder_ = services_->services_provided_to_embedder.Pass();
+    root_bundle_(root_bundle.Pass()),
+    incoming_services_(incoming_services.Pass()) {
+  if (services_ && services_->outgoing_services.is_pending()) {
+    outgoing_services_ = services_->outgoing_services.Pass();
   } else {
-    services_provided_to_embedder_ = GetProxy(&services_from_dart_);
+    outgoing_services_ = GetProxy(&services_from_dart_);
```

Comment:
incoming and outgoing are still pretty confusing terms. it's hard to tell at this point of the code which side of the barrier the code thinks it's on.